openwisp_monitoring/tests/test_selenium.py

```diff
@@ -284,3 +304,165 @@ def test_device_wifi_session_inline_change(self):
                 "innerHTML"
             ),
         )
+
+
+@tag("selenium_tests")
+class TestDashboardMap(
+    SeleniumTestMixin, TestDeviceMonitoringMixin, TestGeoMixin, StaticLiveServerTestCase
+):
+    object_model = Device
+    location_model = Location
+    floorplan_model = Floorplan
+    object_location_model = DeviceLocation
+
+    def test_filtering_on_device_popup(self):
+        d1 = self._create_device(name="Test-Device1", mac_address="00:00:00:00:00:01")
+        d2 = self._create_device(name="Test-Device2", mac_address="00:00:00:00:00:02")
+        d2.monitoring.status = "ok"
+        d2.monitoring.save()
+        location = self._create_location(type="indoor", name="Test-Location")
+        self._create_object_location(
+            content_object=d1,
+            location=location,
+        )
+        self._create_object_location(
+            content_object=d2,
+            location=location,
+        )
+        self.login()
+        self.wait_for(
+            "element_to_be_clickable",
+            By.CSS_SELECTOR,
+            "g path.leaflet-interactive",
+            timeout=5,
+        )
+        location_point = self.find_element(By.CSS_SELECTOR, "g path")
+        location_point.click()
+        self.wait_for_visibility(By.CSS_SELECTOR, ".map-detail")
+        table_entries = self.find_elements(By.CSS_SELECTOR, ".map-detail tbody tr")
+        self.assertEqual(len(table_entries), 2)
+
+        with self.subTest("Test filtering by status"):
```

Comment:
I'd also verify the following things: 

- Multiple filters can be used at the same time 
- The close button appears next to the filter 
- Removing all the filters will hide the close button from all filter and show all the devices.